app/javascript/controllers/header_search_controller.js

```diff
@@ -0,0 +1,10 @@
+import { Controller } from 'stimulus'
+
+export default class extends Controller {
+  
+  submitForm() {
+    const searchForm = document.getElementById('search_form')
```

Comment:
This is what stimulus `targets` are for. You can look at other controllers to pick up the pattern, but essentially you define any element you intend to interact with:

```
static targets = ['form']
```

And then in the html you annotate the element you want connected with `data-target="header-search.form"`

Then you can do `formTarget.submit()`

Reply:
If the controller was on the form instead of the div, you could call `this.element.submit()` (you wouldn't need to go through the hassle of defining an extra target) — however, in this case the image is outside the form, so it makes sense to have the controller on the div IMO.